pkgs/development/libraries/unixODBCDrivers/default.nix

```diff
@@ -119,4 +119,46 @@
       maintainers = with maintainers; [ vlstill ];
     };
   };
+
+  msodbcsql17 = stdenv.mkDerivation rec {
+    name = "msodbcsql17-${version}";
+    version = "${versionMajor}.${versionMinor}.${versionAdditional}-1";
+
+    versionMajor = "17";
+    versionMinor = "2";
+    versionAdditional = "0.1";
+
+    src = fetchurl {
+      url = "https://packages.microsoft.com/debian/9/prod/pool/main/m/msodbcsql17/msodbcsql${versionMajor}_${version}_amd64.deb";
+      sha256 = "1966ymbbk0jsacqwzi3dmhxv2n8hfgnpjsx3hr3n7s9d88chgpx5";
+    };
+
+    buildInputs = [ unixODBC dpkg];
+
+    unpackPhase = "dpkg -x $src ./";
+    buildPhase = "";
+
+    installPhase = ''
+      mkdir -p $out
+      mkdir -p $out/libshim
+      ln -s ${lib.makeLibraryPath [openssl]}/libssl.so.1.0.0 $out/libshim/libssl.so.1.0.2
+      ln -s ${lib.makeLibraryPath [openssl]}/libcrypto.so.1.0.0 $out/libshim/libcrypto.so.1.0.2
+      cp -r opt/microsoft/msodbcsql${versionMajor}/lib64 opt/microsoft/msodbcsql${versionMajor}/share $out/
+      patchelf --set-rpath ${lib.makeLibraryPath [ openssl unixODBC kerberos curl libuuid stdenv.cc.cc]}:$out/libshim \
+        $out/lib64/libmsodbcsql-${versionMajor}.${versionMinor}.so.${versionAdditional}
```

Comment:
Have you tried `autoPatchelfHook`? It might simplify future maintenance of this package: https://github.com/NixOS/nixpkgs/commit/ea5787ad5291ee1c131326cb9c9fec03d359edff

Reply:
Basically it tries to match libraries from `buildInputs` with required libraries in the package.

Reply:
The `openssl` symlink could be added to `$out/lib` so it will be found by the hook.

Reply:
Thanks for the tip!  `autoPatchelfHook` works quite nicely.